### PR TITLE
Refactor multiprocessing for vote

### LINKED ISSUE #494: Port voter to the new pipeline architecture
The goal is to untangle the logic of the voter from the multiprocessing part (that is managing the processes, connect them using queues, and more).

### PR DESCRIPTION
**Pull request by vrde**
Closes #494, #43, #122

### CODE REVIEW

File: bigchaindb/pipelines/vote.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        """
+
+        self.bigchain.write_vote(vote)
+        return vote
+
+
+def initial():
+    """Return unvoted blocks."""
+    b = Bigchain()
+    initial = b.get_unvoted_blocks()
```
Comment (r-marques) [inline bigchaindb/pipelines/vote.py]: confusing assigning a variable with the same name as the method

File: bigchaindb/pipelines/vote.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    """This class encapsulates the logic to vote on blocks.
+
+    Note:
+        Methods of this class will be executed in different processes.
+    """
+
+    def __init__(self):
+        """Initialize the Block voter."""
+
+        last_voted = Bigchain().get_last_voted_block()
```
Comment (r-marques) [inline bigchaindb/pipelines/vote.py]: cleaner if you call this after instantiating `Bigchain`, no need to instantiate it twice

File: bigchaindb/pipelines/vote.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            tx_validity (bool): the validity of the transaction
+            block_id (str): the id of block containing the transaction
+            num_tx (int): the total number of transactions to process
+
+        Returns:
+            None, or a vote if a decision has been reached.
+        """
+
+        self.counters[block_id] += 1
+        self.validity[block_id] = tx_validity and self.validity.get(block_id,
```
Comment (r-marques) [inline bigchaindb/pipelines/vote.py]: Checking that all the transactions are valid is not enough to validate the block.
You also need to check the fields in the block itself, basically call `consensus.validate_block`

**Comment by r-marques** [comment]
This pr is related to #510 
After both are merged there will need to be some cleanup.
I guess `bigchaindb/voter.py` and `tests/db/test_voter.py` could be deleted

File: bigchaindb/pipelines/vote.py (hunk 3 above)
Comment (vrde) [inline bigchaindb/pipelines/vote.py]: You are right, will update my PR to check it!

File: bigchaindb/pipelines/vote.py (hunk 2 above)
Comment (vrde) [inline bigchaindb/pipelines/vote.py]: It's to overcome a limitation with `multiprocessing`. Will add a comment to explain that :)

**Comment by r-marques** [comment]
:+1:

**Pull request merged**